hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -8408,42 +8423,21 @@ public void startRegionOperation() throws IOException {
 
   @Override
   public void startRegionOperation(Operation op) throws IOException {
-    boolean isInterruptableOp = false;
-    switch (op) {
-      case GET: // interruptible read operations
-      case SCAN:
-        isInterruptableOp = true;
-        checkReadsEnabled();
-        break;
-      case INCREMENT: // interruptible write operations
-      case APPEND:
-      case PUT:
-      case DELETE:
-      case BATCH_MUTATE:
-      case CHECK_AND_MUTATE:
-        isInterruptableOp = true;
-        break;
-      default: // all others
-        break;
-    }
-    if (
-      op == Operation.MERGE_REGION || op == Operation.SPLIT_REGION || op == Operation.COMPACT_REGION
-        || op == Operation.COMPACT_SWITCH
-    ) {
-      // split, merge or compact region doesn't need to check the closing/closed state or lock the
-      // region
-      return;
-    }
+    boolean isInterruptibleOp = isOperationInterruptible(op);
```

Comment:
I think it's weird for a method called `isOperationInterruptible` to throw an exception. I guess one can inspect the method and follow the logic, but it still smells weird. I would probably factor the exception throwing code into a new method `checkOperationAllowed()` or something similar, and call it separately before calling `isOperationInterruptible`.

(Or alternatively revert to the original format - the method was 60 lines which isn't extremely long relatively speaking... though I like your general move towards smaller methods and less cognitive load in reading those methods.)

Reply:
> Or alternatively revert to the original format - the method was 60 lines which isn't extremely long relatively speaking...

+1 to this. The refactor is good but it would look better as separate PR for the change tracking purpose, specifically helps when we refactor critical piece of the code.